main.nf

```diff
@@ -167,46 +486,708 @@ process fastqc {
+process kraken2 {
+    tag "${name}-${db_name}"
+    publishDir "${params.outdir}/Taxonomy/kraken2/${name}", mode: 'copy',
+            saveAs: {filename -> filename.indexOf(".krona") == -1 ? filename : null}
+
+    input:
+    set val(name), file(reads), val(db_name), file("database/*") from kraken2_input
+
+    output:
+    set val("kraken2"), val(name), file("results.krona") into kraken2_to_krona
+    file("kraken2_report.txt")
+
+    script:
+    def input = params.singleEnd ? "\"${reads}\"" :  "--paired \"${reads[0]}\" \"${reads[1]}\""
+    """
+    kraken2 \
+        --report-zero-counts \
+        --threads "${task.cpus}" \
+        --db database \
+        --fastq-input \
+        --report kraken2_report.txt \
+        $input \
+        > kraken2.kraken
+    cat kraken2.kraken | cut -f 2,3 > results.krona
+    """
+}
+
+process krona_db {
+    output:
+    file("taxonomy/taxonomy.tab") into file_krona_db
+
+    when:
+    ( params.centrifuge_db || params.kraken2_db ) && !params.skip_krona
+
+    script:
+    """
+    ktUpdateTaxonomy.sh taxonomy
+    """
+}
+
+centrifuge_to_krona
+    .mix(kraken2_to_krona)
+    .combine(file_krona_db)
+    .set { krona_input }
+
+process krona {
+    tag "${classifier}-${name}"
+    publishDir "${params.outdir}/Taxonomy/${classifier}/${name}", mode: 'copy'
+
+    input:
+    set val(classifier), val(name), file(report), file("taxonomy/taxonomy.tab") from krona_input
+
+    output:
+    file("*.html")
+
+    script:
+    """
+    ktImportTaxonomy "$report" -tax taxonomy
+    """
+}
+
+
+/*
+ * STEP 2 - Assembly
+ */
+process megahit {
+    tag "$name"
+    publishDir "${params.outdir}/", mode: 'copy',
+        saveAs: {filename -> filename.indexOf(".fastq.gz") == -1 ? "Assembly/$filename" : null}
+
+    input:
+    set val(name), file(reads) from trimmed_reads_megahit
+
+    output:
+    set val("MEGAHIT"), val("$name"), file("MEGAHIT/${name}.contigs.fa") into assembly_megahit_to_quast
+    set val("MEGAHIT"), val("$name"), file("MEGAHIT/${name}.contigs.fa") into assembly_megahit_to_metabat
+    file("MEGAHIT/*.log")
+
+    when:
+    !params.skip_megahit
+
+    script:
+    def input = params.singleEnd ? "-r \"${reads}\"" :  "-1 \"${reads[0]}\" -2 \"${reads[1]}\""
+    """
+    megahit -t "${task.cpus}" $input -o MEGAHIT --out-prefix "${name}"
+    """
+}
+
+
+/*
+ * metaSpades hybrid Assembly
+ */
+
+ files_lr_filtered
+    .combine(trimmed_sr_spadeshybrid, by: 0)
+    .set { files_pre_spadeshybrid }
+
+process spadeshybrid {
+    tag "$id"
+    publishDir "${params.outdir}/", mode: 'copy', pattern: "${id}*",
+        saveAs: {filename -> filename.indexOf(".fastq.gz") == -1 ? "Assembly/SPAdesHybrid/$filename" : null}
+
+    input:
+    set id, file(lr), file(sr) from files_pre_spadeshybrid  
+
+    output:
+    set id, val("SPAdesHybrid"), file("${id}_graph.gfa") into assembly_graph_spadeshybrid
+    set val("SPAdesHybrid"), val("$id"), file("${id}_scaffolds.fasta") into assembly_spadeshybrid_to_quast
```

Comment:
Drop the `${id}_` if you're not doing the `cp` / `mv` of course.

Also it's slightly confusing that in the three channels you're doing:

* id, tool, file
* tool, id, file
* tool, id, file

I usually try to keep the order of arguments consistent, for my own sanity 😆 

Note that we should now be using `path` instead of `file` I think..